cdc/scheduler/internal/v3/replication/replication_manager.go

```diff
@@ -573,6 +573,13 @@ func (r *Manager) AdvanceCheckpoint(
 		return true
 	})
 	if cannotProceed {
+		if redoMetaManager.Enabled() {
```

Comment:
Is it possible to abstract the redo processing logic (L576 and L634) into a function?